src/Reflection/ReflectionClass.php

```diff
@@ -418,54 +377,69 @@ private function getMethodsFromInterfaces(AlreadyVisitedClasses $alreadyVisitedC
      */
     private function getMethodsIndexedByLowercasedName(AlreadyVisitedClasses $alreadyVisitedClasses): array
     {
-        $alreadyVisitedClasses->push($this->getName());
-
-        if ($this->cachedMethods !== null) {
-            return $this->cachedMethods;
-        }
```

Comment:
aren't you missing this? it currently looks like the cache is only written, but never read

Reply:
@herndlm Yes, it's already fixed :) Probably removed during rebase by accident.